framework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,272 @@
+    for child in paramInput.subparts:
+      if child.getName() == 'featureParameters':
+        self.featureParameters = child.value
+      elif child.getName() == 'targetParameters':
+        self.targetParameters = child.value
+      elif child.getName() == 'targetPivotParameter':
+        self.targetPivotParameter = child.value
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it loads the
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    dataSets = [data for _, _, data in inputIn['Data']]
+    pivotParameter = self.pivotParameter
+    names=[]
+    if isinstance(inputIn['Data'][0][-1], xr.Dataset):
+      names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+      # names = [inp[-1].attrs['name'] for inp in inputIn['Data']]
+      if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+        if 'dynamic' not in self.dynamicType: #self.model.dataType:
+          self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+        else:
+          pivotParameter = self.pivotParameter
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataSets, **{'dataobjectNames': names}).items()}#inputIn
+    # if pivotParameter:
+    #   # Uncomment this to cause crash: print(dataSets[0], pivotParameter)
+    #   if len(dataSets[0][pivotParameter]) != len(list(evaluation.values())[0]):
+    #     self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+    #   if pivotParameter not in evaluation:
+    #     evaluation[pivotParameter] = dataSets[0][pivotParameter]
+    return evaluation
+
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    # self.stat.run({'targets':{self.target:xr.DataArray(self.functionS.evaluate(tempDict)[self.target])}})[self.computationPrefix +"_"+self.target]
+    senMeasurables = self.stat.run({"Data":[[None, None, datasets[0]]]})
+    senFOMs = self.stat.run({"Data":[[None, None, datasets[1]]]})
+
+
+    # for data in datasets:
+    #   sen = self.stat.run(data)
+    names = kwargs.get('dataobjectNames')
+    outs = {}
+    # for feat, targ, param, targParam in zip(self.features, self.targets, self.Parameters, self.targetParameters):
+    for feat, targ, param, targParam in zip(self.features, self.targets, self.featureParameters, self.targetParameters):
```

Comment:
Do you assume the length of features, targets, featureParameters, targetParameters are the same? If so, I think you better to check it handleInput or initialization method

Reply:
No they shouldn't be.